bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/handler/FreeAtHomeBridgeHandler.java

```diff
@@ -0,0 +1,832 @@
+        return channelUpdateHandler;
+    }
+
+    /**
+     * Method to process socket events
+     */
+    public void processSocketEvent(String receivedText) {
+        JsonReader reader = new JsonReader(new StringReader(receivedText));
+        reader.setLenient(true);
+        JsonElement jsonTree = JsonParser.parseReader(reader);
+
+        // check the output
+        if (jsonTree.isJsonObject()) {
+            JsonObject jsonObject = jsonTree.getAsJsonObject();
+
+            jsonObject = jsonObject.getAsJsonObject(sysApUID);
+            jsonObject = jsonObject.getAsJsonObject("datapoints");
+
+            Set<String> keys = jsonObject.keySet();
+
+            Iterator<String> iter = keys.iterator();
+
+            while (iter.hasNext()) {
+                String eventDatapointID = iter.next();
+
+                JsonElement element = jsonObject.get(eventDatapointID);
+                String value = element.getAsString();
+
+                channelUpdateHandler.updateChannelByDatapointEvent(eventDatapointID, value);
+
+                logger.debug("Socket event processed: event-datapoint-ID {} value {}", eventDatapointID, value);
+            }
+        }
+    }
+
+    /**
+     * Method to open Http connection
+     */
+    public boolean openHttpConnection() {
+        boolean ret = false;
+
+        // Configure client
+        httpClient.setFollowRedirects(false);
+        httpClient.setMaxConnectionsPerDestination(1);
+        httpClient.setMaxRequestsQueuedPerDestination(50);
+
+        // Set timeouts
+        httpClient.setIdleTimeout(-1);
+        httpClient.setConnectTimeout(5000);
+
+        try {
+            // Start HttpClient.
+            switch (httpClient.getState()) {
+                case AbstractLifeCycle.FAILED:
+                case AbstractLifeCycle.STOPPED: {
+                    httpClient.start();
+                    break;
+                }
+                case AbstractLifeCycle.STARTING:
+                case AbstractLifeCycle.STARTED:
+                case AbstractLifeCycle.STOPPING: {
+                    // nothing to do
+                    break;
+                }
+            }
+
+            logger.debug("Start http client");
+
+            ret = true;
+        } catch (Exception ex) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "@text/comm-error.not-able-start-httpclient");
+
+            ret = false;
+        }
+
+        if (ret) {
+            // Add authentication credentials and make a check.
+            try {
+                // Add authentication credentials.
+                AuthenticationStore auth = httpClient.getAuthenticationStore();
+
+                URI uri1 = new URI("http://" + ipAddress + "/fhapi/v1");
+                auth.addAuthenticationResult(new BasicAuthentication.BasicResult(uri1, username, password));
+
+                String url = "http://" + ipAddress + "/fhapi/v1/api/rest/devicelist";
```

Comment:
Can `baseUrl` be used here to avoid this duplication? This is already initialized in `initialize`:
```java
            // build base URL
            baseUrl = "http://" + ipAddress + "/fhapi/v1/api";
```

So I guess this should work:
```suggestion
                String url = baseUrl + "/rest/devicelist";
```